TESTS/mbedmicro-rtos-mbed/signals/main.cpp

```diff
@@ -15,49 +15,357 @@
  */
 #include "mbed.h"
 #include "greentea-client/test_env.h"
+#include "unity.h"
+#include "utest.h"
 #include "rtos.h"
 
+using namespace utest::v1;
+
+
 #if defined(MBED_RTOS_SINGLE_THREAD)
   #error [NOT_SUPPORTED] test not supported
 #endif
 
-#define TEST_STACK_SIZE 512
+#define TEST_STACK_SIZE   512
+#define ALL_SIGNALS 0x7fffffff
+#define PROHIBITED_SIGNAL 0x80000000
+#define MAX_FLAG_POS 30
+
+Semaphore sem_a(0, 1);
+Semaphore sem_b(0, 1);
+volatile int32_t ret_val;
+volatile osStatus ret_status;
+
+
+template <int32_t signals, uint32_t timeout>
+void run_signal_wait(void)
+{
+    osEvent ev = Thread::signal_wait(signals, timeout);
+    ret_status = ev.status;
```

Comment:
Why don't we assert correct status here? But we pass it back to main thread?

Reply:
I assumed that asserting in main function will make it easier to analyze the tests

Reply:
I think it works both ways, with 'where did that value came from' factor. @0xc0170 any preferences from your side?

Reply:
My preference would be the failure as soon as it happens (in this scope in this case)